### DISCUSSION TITLE
Chainlit integration

### DISCUSSION THREAD

**Discussion by mlops-v2-gs** [sinaptik-ai/pandas-ai#648] [General]
Hi,
Is there a chainlit parser or method to show the charts (and other responses) generated by your library?
As a workaround, at the moment I save the image and show it within the chainlit but there is a plot window opens every time.  (i tried open_charts: False in config as suggested elsewhere too) 
Any suggestions which I can try?
Thanks in advance.

**Comment by gventuri** [comment]
Hi @mlops-v2-gs did you check the streamlit parser? Maybe with a few changes it could work also for chainlit.

**Reply by mlops-v2-gs** [reply]
Hi @gventuri  thanks for this. I did try that one, but it failed to suppress the pop-up plot window.

**Comment by imranrazakhan** [comment]
With Pandasai, i was able to generate response like below

```
class ChainlitOuput(ResponseParser):
    def __init__(self, context) -> None:
        super().__init__(context)
    def format_dataframe(self, result):
        print("format_dataframe")
        return
    def format_plot(self, result):
        image = cl.Image(path=result["value"], name="image1", display="inline")
        return image
    def format_other(self, result):
        print("format_other")
        return
```
Pandasai.Agent

```
agent = Agent([df],
              config={"llm": local_llm, "save_charts": True, "verbose": True, "response_parser": ChainlitOuput})
```

Show message with below code

```
     agent_response = agent.chat(message.content)
    await cl.Message(
        content="This message has an image!",
        elements=[agent_response],
    ).send()
```
